### ISSUE TITLE
Editor button doesn't properly update after bind or unbind

### ISSUE THREAD

**Issue by Kasterisk** [Revolutionary-Games/Thrive#2521]
If I recall correctly this is already fixed before(?) but apparently the bug is still alive and kicking. On bind, the button is still flashing and if you press it, the transition starts but it only realizes the player is bounded after the transition is finished (aborts the scene switch), so there seems to be a guard internally but no immediate update to the editor button. Likewise for when unbinding.

**Comment by hhyyrylainen** [comment]
Yes, the button state doesn't update correctly. After this is fixed, the extra guard before the scene transition can (probably) be removed as it should be unnecessary then.

**Comment by Kasterisk** [comment]
Ah okay, just realized that this is actually part of #2504. It's still good to have a dedicated issue I guess.

**Comment by Maxonovien** [comment]
I think that PR #2390 was supposed to fix it. I'll try to see why it seems not to work anymore

**Comment by Maxonovien** [comment]
Okay, I found what went wrong: 

I fixed it earlier by adding a line at colony creation (instanciation) to tell the master cell that it was added to a colony.
This line was removed in the following commit from @Kemikal1 : https://github.com/Kemikal1/Thrive/commit/2ed8f4b7da5bf0bc61db349be2065f4df4c0a34b#diff-d37f14b70e271e446c8dd3a1ada2f7bb70660e4c45af97579062481e33b051a6
This probably brought the bug back again, but a safeguard was implemented in the mean time.

However, the method `OnColonyMemberAdded` was modified and now uses a reference to the colony so it can no longer be called from the constructor. Either we go back on this, or we add a call to `OnIGotAddedToColony`(or `OnColonyMemberAdded(this)`) after every call to the constructor (only one at this point, but it may change and we'd have to keep that in mind; but it allows for decoupling if necessary).
